linera-execution/src/runtime.rs

```diff
@@ -68,6 +71,34 @@ struct ApplicationStatus {
     signer: Option<Owner>,
 }
 
+/// A loaded application instance.
+#[derive(Debug)]
+struct LoadedApplication<Instance> {
+    instance: Arc<Mutex<Instance>>,
+    parameters: Vec<u8>,
+}
+
+impl<Instance> LoadedApplication<Instance> {
+    /// Creates a new [`LoadedApplication`] entry from the `instance` and its `description`.
+    fn new(instance: Instance, description: UserApplicationDescription) -> Self {
+        LoadedApplication {
+            instance: Arc::new(Mutex::new(instance)),
+            parameters: description.parameters,
+        }
+    }
+}
+
+impl<Instance> Clone for LoadedApplication<Instance> {
+    // Manual implementation is needed to prevent the derive macro from adding an `Instance: Clone`
+    // bound
```

Comment:
The comment is quite optional IMHO.

Reply:
This is mostly a reminder for me, because more than once I've got distracted upon seeing a manual `Clone` implementation to think if it could have been derived instead :sweat_smile: